pandas/core/arrays/categorical.py

```diff
@@ -2536,9 +2304,7 @@ def _replace(self, *, to_replace, value, inplace: bool = False):
                 continue
             if replace_value in cat.categories:
                 if isna(new_value):
-                    with catch_warnings():
-                        simplefilter("ignore")
-                        cat.remove_categories(replace_value, inplace=True)
+                    cat = cat.remove_categories(replace_value)
```

Comment:
i think we need/want to do something here to avoid constructing a new object at each point in the loop

Reply:
updated to operate inplace within the loop